src/panels/lovelace/editor/card-editor/hui-dialog-edit-card.ts

```diff
@@ -32,20 +33,23 @@ export class HuiDialogEditCard extends LitElement {
 
   @property() private _cardConfig?: LovelaceCardConfig;
 
+  @property() private _newCard?: boolean;
+
   constructor() {
     super();
     this._cardPicked = this._cardPicked.bind(this);
     this._cancel = this._cancel.bind(this);
+    this._save = this._save.bind(this);
   }
 
   public async showDialog(params: EditCardDialogParams): Promise<void> {
     this._params = params;
-    this._cardConfig =
-      params.path.length === 2
-        ? (this._cardConfig = params.lovelace.config.views[
-            params.path[0]
-          ].cards![params.path[1]])
-        : undefined;
+    this._newCard = params.path.length !== 2;
+    this._cardConfig = !this._newCard
+      ? (this._cardConfig = params.lovelace.config.views[params.path[0]].cards![
+          (params as any).path[1]
```

Comment:
why do we have to add `as any` here ?

Reply:
Because `views` has the type `[number] | [number, number]`, and undefined can't be an index.

Previously, typescript was smart enough to see the params.path.length === 2 check and underst... maybe it works if I change it to `this,_newCard = !(params.path.length === 2)` ...

Reply:
To filter out `undefined`, we can either add a `!` or do a `if (!params.path)` to teach the TS compiler it's not undefined.

Reply:
The problem is the second element specifically.
Since `params.path` is know no be defined, so is `params.path[0]`, but not `params.path[1]`.

I couldn't find the correct magic incantation to force that...

Reply:
Maybe `(params.path as [number, number])[1]` would be clearer.

Reply:
TypeScript will only know it contains length 2 if we check for that. It is not aware that `this._newCard` contains that information.

Reply:
I think I came up with a nicer way of doing it.